src/main/java/org/sagebionetworks/bridge/dao/HealthDataEx3Dao.java

```diff
@@ -0,0 +1,31 @@
+package org.sagebionetworks.bridge.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecordEx3;
+
+/**
+ * Exporter 3 version of the Health Data DAO. Methods in this interface only apply to the EX3 implementation of Health
+ * Data Records, not to the original implementation of Health Data records.
+ */
+public interface HealthDataEx3Dao {
+    /** Create or update health data record. Returns the created or updated record. */
+    HealthDataRecordEx3 createOrUpdateRecord(HealthDataRecordEx3 record);
+
+    /** Deletes all health data records for the given health code. */
+    void deleteRecordsForHealthCode(String healthCode);
+
+    /** Retrieves the record for the given ID. */
+    Optional<HealthDataRecordEx3> getRecord(String id);
+
+    /** Retrieves all records for the given healthcode and time range. */
+    List<HealthDataRecordEx3> getRecordsForHealthCode(String healthCode, long createdOnStart, long createdOnEnd);
+
+    /** Retrives all records for the given app and time range. */
+    List<HealthDataRecordEx3> getRecordsForApp(String appId, long createdOnStart, long createdOnEnd);
```

Comment:
This doesn't have to be paged? (Same for the other ones, I'm assuming your limiting the records with the time range, but if we had a super-busy app, could this get overwhelmed?)